primefaces-integration-tests/src/test/java/org/primefaces/integrationtests/treetable/AbstractTreeTableTest.java

```diff
@@ -113,4 +116,18 @@ protected void assertMessage(Messages messages, int index, String summary, Strin
         Assertions.assertEquals(summary, message.getSummary());
         Assertions.assertEquals(detail, message.getDetail());
     }
+
+
+    protected void filterGlobal(InputText inputGlobalFilter, String filter) {
+        /*
+         * TreeTable-filter ist triggered by keyup. (filterEvent is not configureable like for DataTable)
+         */
+
+        // maybe we can move some of this to PF Selenium (InputText?, TreeTable?)
+        inputGlobalFilter.clear();
+//        PrimeSelenium.guardAjax(inputGlobalFilter).sendKeys(filter); // does not work - causes org.openqa.selenium.TimeoutException: Timeout while waiting for AJAX complete!
+        inputGlobalFilter.sendKeys(filter);
+        PrimeSelenium.guardAjax(inputGlobalFilter).sendKeys(Keys.TAB);
+        PrimeSelenium.wait(500); // ugly work-around because PrimeSelenium.guardAjax does not work
```

Comment:
Better ideas/solutions are welcome.

Reply:
could you add some logging to see what happens?
add start and end js logging in the guard and also catch PF ajax events with jquery and log it?

so we can see if the guard is around the ajax request or happens before

Reply:
same for the other open PR

Reply:
Look´s like treetable-global-filter fires 2 ajax-request (when controlled via Selenium?)
![image](https://user-images.githubusercontent.com/10461942/124969537-1d9b5e80-e027-11eb-8471-ec614c389675.png)

The guard works right (for the first ajax-request)
![image](https://user-images.githubusercontent.com/10461942/124969603-30159800-e027-11eb-8c7d-8b2ba135bd7a.png)

But probably the second ajax-request causes timing-trouble.

Reply:
That´s our "bad guy":
![image](https://user-images.githubusercontent.com/10461942/124972630-e62eb100-e02a-11eb-8269-43945996d8ec.png)